### ISSUE TITLE
Form validation catching all exceptions and re-raising as ValidationError

### ISSUE THREAD

**Issue by tomage** [jyveapp/django-args#5]
I recently found that a django-action-framework action (which is baed on `django-args`) wasn't behaving as expected, and it was due to an error in the code that we should have caught in unit-tests, or at least in QA stages, but we just didn't have proper coverage of exactly that situation.

In writing a failing unit-test I noticed something odd. I was getting a 200 HTTP response from the action, where I was expecting very much to get a stacktrace, or at least a 500 error.

Realized that the form cleaning in `djarg.forms` catches all exceptions [(link to code)](https://github.com/jyveapp/django-args/blob/master/djarg/forms.py#L16) and re-raises as ValidationErrors (from Django), which ended up, in my case, as a 200 HTTP response.

I figured, perhaps we shouldn't catch all exceptions, and rather let actual bugs - like IndexError, or even NameError - bubble up to the end-user. It'll of course usually result in a server error. But one might argue that'd be better than showing the end-user a technical error via the messages framework, without any notification to the engineers.

It is a bit harsher way of doing this, and to handle certain errors more gracefully one would have to capture particular exceptions in the validation/clean methods, but I think that might be preferred.

I looked to see if I could "simply" override this on my end, and I believe I can do that, but I figured it might be worth raising this here.

Perhaps this could be an opt-in feature too.. Either global settings variable (i.e. `DJARG_CAPTURE_ALL_EXCEPTIONS_AS_VALIDATION_ERRORS`), or even a per-form/per-view setting, to opt-in (or out) of capturing of exceptions like this. Or even list exceptions one would like to be captured as ValidationErrors (for example `ValueError` which one could argue _usually_ indicates bad user input, tho not always).

Anyway.. I opened a very similar argument against `django-action-framework` as well, so I suppose similar arguments for/against work for both cases, but def. want to get your views on this @wesleykendall  :)

**Comment by tomage** [comment]
Hm.. scratch that.. I found that I misread something - and that my test I used was faulty. Closing for now.. Tho there's something odd going on I still have yet to figure out.

**Comment by tomage** [comment]
No - nevermind - I was on track the first time lol. (I've been staring at too much code all day long).

**Comment by wesleykendall** [comment]
@tomage one way that we can aim to solve these issues is also by making all of these exception based context managers take the same named argument. I.e. if this context manager took a ``base_handled_exception_class`` argument, along with all other context managers in django-action-framework, one only needs to then pass in a `base_handled_exception` argument to the called function and it will then be used throughout. It's probably going to be easiest to approach it this way and then supply `base_handled_exception` as one of the `default_args` of https://github.com/jyveapp/django-args/blob/master/djarg/views.py#L185. This variable could either be supplied as a setting or a class variable too (although I'd argue one can just make a base class in their app that overrides this method to supply it)

**Comment by tomage** [comment]
Yeah - I was thinking along similar lines RE: usage of settings at least. And makes sense to pass along via `default_args` as you suggest.

So - were you thinking in the style of how Django does things when overriding classes in settings, something like:

```
# In settings.py
DJANGO_ARGS_BASE_HANDLED_EXCEPTION_CLASS = 'myproject.exceptions.MyValidationError'
```
or something like that?
I.e. the library would then import that object, and insert it as the exception class to use as `base_handled_exception` in the view's `default_args()` method?

I would probably set it as `django.core.exceptions.ValidationError` in my case - tho it would be nice to be able to override per view (i.e. via overriding `default_args()`).

Is this something you'd want to execute on, or, if not, I might be able to grab someone from Jyve to try their hand at it.
